notebooks/cva_opt_with_channels_example.ipynb

```diff
@@ -0,0 +1,435 @@
+   "source": [
+    "Please note that CVA_OPT will use max_generation * lambda many function evaluations, i.e.,\n",
+    "50 * 10 = 500 function evaluations to minimize the Rosenbrock function."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e968b38b-f7ab-4fcd-acb9-b31568386c8a",
+   "metadata": {},
+   "source": [
+    "### Set up channel and callback function to evaluate the objective function\n",
+    "CVA_OPT is a black-box optimizer using evolutionary strategies (ES) meaning it does not know anything \n",
+    "about the internal structure of the objective function which should be optimized. CVA_OPT sends \n",
+    "candidate solutions to a so called channel implementing an ask and tell interface with a callback function. The channel sends the candidate solutions. There are evaluated by the callback function, i.e., assigning for each candidate the objective value. Then CVA_OPT reads the result from the channel and the next iteration is started."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2a4f5f37-be09-4d6f-8eda-fd1316cebba3",
+   "metadata": {},
+   "source": [
+    "#### Define a function to run an experiment with channel\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "31e66e09-6761-48d1-8580-f4425324a1cd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "async def run_exp_with_channel(client, experiment_id, cva_params):\n",
+    "    # set up channel\n",
+    "    channel = await Channel.create(client=client)\n",
+    "    # extend cva_params with func_eval_worker_channel_id\n",
+    "    cva_params['func_eval_worker_channel_id'] = channel.id\n",
+    "    \n",
+    "    # create the experiment run\n",
+    "    run = ExperimentRun.create(\n",
+    "        client=client, \n",
+    "        algorithm=AlgorithmProto.CVA_OPT, \n",
+    "        experiment_id=experiment_id,\n",
+    "        compute=HardwareProto.SMALL,\n",
+    "        parameters={'parameters': cva_params, 'inputs': {} }\n",
+    "    )\n",
+    "    print(f\" starting run with id {run.id}\")\n",
+    "\n",
+    "    # define the callback function\n",
+    "    async def callback(ask: Ask) -> Tell:\n",
+    "        dim = len(ask.headers)\n",
+    "        func = otf.Rosenbrock(dim)\n",
+    "        y = []\n",
+    "        for parameter in ask.parameters:\n",
+    "            y.append(float(func(np.array(parameter.values))))\n",
+    "        # add result to data\n",
+    "\n",
+    "        return Tell(\n",
+    "            parameters=ask.parameters,\n",
+    "            headers=ask.headers,\n",
+    "            results=y\n",
+    "        )\n",
+    "    # define a function to be called after the optimization is finished\n",
+    "    def success():\n",
+    "        pass\n",
+    "\n",
+    "    # let the channel wait for connections\n",
+    "    is_finished = False\n",
+    "    max_retries = 10\n",
+    "    retries = 0\n",
+    "    while (not is_finished) and (retries < max_retries):\n",
```

Comment:
next week with the new feature I will be working on this should get way better. But for now I would be fine with doing this workaround.